Ginger/Ginger/ApplicationModelsLib/APIModels/APIModelWizard/AddAPIModelSelectTypePage.xaml.cs

```diff
@@ -388,20 +400,170 @@ private void BrowseForTemplateFiles(string files = null)
             {
                 foreach (String file in dlg.FileNames)
                 {
-                    AddAPIModelWizard.XTFList.Add(new TemplateFile() { FilePath = file });
+                    if (await ValidateFile(file))
+                    {
+                        AddAPIModelWizard.XTFList.Add(new TemplateFile() { FilePath = file });
+                    }
+                }
+            }
+        }
+
+        private async Task<bool> ValidateFile(string fileName = "")
+        {
+            bool bIsFileValid = false;
+            try
+            {
+                if (APITypeComboBox.SelectedValue.ToString() == eAPIType.WSDL.ToString())
+                {
+                    fileName = xURLTextBox.Text;
+                    bIsFileValid = await LoadWSDLFileValidation();
+                }
+                else if (APITypeComboBox.SelectedValue.ToString() == eAPIType.Swagger.ToString())
+                {
+                    SwaggerParser swagger = new SwaggerParser();
+                    ObservableList<ApplicationAPIModel> swaggerList = new ObservableList<ApplicationAPIModel>();
```

Comment:
Please use plain list instead of observablelist if it is not binded with any UI element